server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -686,12 +701,21 @@ private static final class ResourceHandlingHttpChannel implements RestChannel {
         private final RestChannel delegate;
         private final CircuitBreakerService circuitBreakerService;
         private final int contentLength;
+        private final MethodHandlers methodHandlers;
+        private final long startTime;
         private final AtomicBoolean closed = new AtomicBoolean();
 
-        ResourceHandlingHttpChannel(RestChannel delegate, CircuitBreakerService circuitBreakerService, int contentLength) {
+        ResourceHandlingHttpChannel(
+            RestChannel delegate,
+            CircuitBreakerService circuitBreakerService,
+            int contentLength,
+            MethodHandlers methodHandlers
+        ) {
             this.delegate = delegate;
             this.circuitBreakerService = circuitBreakerService;
             this.contentLength = contentLength;
+            this.methodHandlers = methodHandlers;
+            this.startTime = System.currentTimeMillis();
```

Comment:
This clock is not monotonic, we should be using `System::nanoTime` (or, better, `ThreadPool#rawRelativeTimeInMillis`)

Reply:
ThreadPool is not available in RestController. So I chose to use `System::nanoTime`. In fact, I copied what `ThreadPool#rawRelativeTimeInMillis` does. This method can potentially be static and reused here. But it seems some tests rely on it being an instance method so that it can be overriden. So I decided to just copy it for now since the duplication is minimal. 
Please refer to 4a4cd4a87df3ce8f865abfa4e7fb3aca315b199a